src/aleph/vm/orchestrator/views/operator.py

```diff
@@ -197,7 +197,7 @@ async def operate_erase(request: web.Request, authenticated_sender: str) -> web.
 
     # Stop the VM
     await pool.stop_vm(execution.vm_hash)
-    await pool.forget_vm(execution.vm_hash)
+    pool.forget_vm(execution.vm_hash)
```

Comment:
I am wondering whether this makes sense here... :thinking: 

We use an event, `VmExecution.stop_event`, to automatically forget a VM once it has stopped.
See https://github.com/aleph-im/aleph-vm/blob/0.4.0a3/src/aleph/vm/pool.py#L131-L135

The line above, `await pool.stop_vm(execution.vm_hash)`, should await `execution.stop()`, which ends with `self.stop_event.set()`, triggering the forget.

Is there an issue in that logic ?

Reply:
This event is only used for ephemeral VMs, because for permanent we don't use anymore if we restart the orchestrator, because is called on `create_a_vm` that implements a task to call `forget_on_stop` when it's stopped.
To improve it, we can implement the `forget_on_stop` method to the entire pool receiving an extra `vm_hash` argument, and create the task on the `VMExecution` instance creation.
Anyway, the `forget_vm` method has a try catch if the execution doesn't exist, so will never fail.